client/grpc/grpc.go

```diff
@@ -42,26 +67,36 @@ func (c *headersCarrier) Set(key, val string) {
 	c.Ctx = metadata.AppendToOutgoingContext(c.Ctx, key, val)
 }
 
-func unaryInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn,
-	invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
-	span, ctx := trace.ChildSpan(ctx, trace.ComponentOpName(componentName, method), componentName, ext.SpanKindProducer,
-		ext.SpanKindProducer)
+func unaryInterceptor(target string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		span, ctx := trace.ChildSpan(ctx,
+			trace.ComponentOpName(componentName, method),
+			componentName,
+			ext.SpanKindProducer,
+		)
+		carrier := headersCarrier{Ctx: ctx}
+		err := span.Tracer().Inject(span.Context(), opentracing.TextMap, &carrier)
+		if err != nil {
```

Comment:
@mantzas Do we really want to return here, if tracing headers cannot be injected, without invoking the RPC?

Reply:
Good point. This error should obviously never happen but maybe we should just log the error and continue.